docs/reference.rst

```diff
@@ -0,0 +1,58 @@
+Definition Reference
```

Comment:
Looking really good! My only comment looking over the structure would be the title of the section, "Definition Reference" seems jargon-y, I'd image few people would understand it off the bat. Even calling it "Execution Environment Definition" could help. Or "Building Inputs to Builder" or "How to Define an Execution Environment"

Reply:
I was actually thinking the same thing! 

![](https://media.giphy.com/media/PmiVw7JZ2Z9K4c4BZy/giphy.gif)